### PR TITLE
6659 use lines for over and undershoot

### LINKED ISSUE #6659: Limiter graph: use lines to represent overshoot and undershoot
At the moment we are using areas for that. @Tantacrul brought forward convincing arguments why this should be using lines instead. The goal here is to improve intelligibility of the graph.

### PR DESCRIPTION
**Pull request by saintmatthieu**
Resolves: #6659

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

File: src/effects/CompressionMeterPanel.cpp [hunk 1]
```diff
@@ -43,19 +43,23 @@ EVT_TIMER(timerId, CompressionMeterPanel::OnTimer)
...
-    , mPlaybackStartStopSubscription {
-       static_cast<InitializeProcessingSettingsPublisher&>(instance).Subscribe(
-          [&](const std::optional<InitializeProcessingSettings>& evt) {
-             if (evt)
-                Reset();
-             else
-                mStopWhenZero = true;
-          })
-    }
+    , mPlaybackStartStopSubscription { static_cast<
```
Comment (grliszas14) [inline src/effects/CompressionMeterPanel.cpp:50]: nitpick: did the autoformatter do it? It was way easier to read before

File: src/effects/DynamicRangeProcessorEditor.cpp [hunk 2]
```diff
@@ -19,7 +19,7 @@
 #include <wx/textctrl.h>
 
 #if wxUSE_ACCESSIBILITY
-#include "WindowAccessible.h"
+#   include "WindowAccessible.h"
```
Comment (grliszas14) [inline src/effects/DynamicRangeProcessorEditor.cpp:22]: unintentional change?

**Review by grliszas14 — APPROVED** [review]

**Review by DavidBailes — APPROVED** [review]
The accessibility changes for the changes to the check boxes are fine.

File: src/effects/CompressionMeterPanel.cpp (hunk 1 above)
Comment (saintmatthieu) [reply src/effects/CompressionMeterPanel.cpp:50]: Yeah, this must be a spec of the `.clang-format` you can find in some of our subdirectories.

File: src/effects/DynamicRangeProcessorEditor.cpp (hunk 2 above)
Comment (saintmatthieu) [reply src/effects/DynamicRangeProcessorEditor.cpp:22]: Same reason as above, but that one looks peculiar all right ...

File: src/effects/DynamicRangeProcessorHistoryPanel.cpp [hunk 3]
```diff
@@ -37,8 +37,18 @@ static const wxColor inputColor { 71, 154, 42, 128 };
 static const wxColor outputColor { 103, 124, 228, 128 };
 static const wxColor actualCompressionColor { 255, 255, 255 };
 
-constexpr auto actualCompressionLineWidth =
-   DynamicRangeProcessorPanel::targetCompressionLineWidth + 4;
+bool MayUsePenGradients()
+{
+   // MacOS doesn't cope well with pen gradients. (Freezes in debug and is
```
Comment (grliszas14) [inline src/effects/DynamicRangeProcessorHistoryPanel.cpp:42]: nitpick: you can write "MacOS and Linux doesn't cope..." as this function checks if the current platform is running Windows (only) -  just to to reduce potential questions in the future

**Review by grliszas14 — APPROVED** [review]

File: src/effects/DynamicRangeProcessorHistoryPanel.cpp (hunk 3 above)
Comment (saintmatthieu) [reply src/effects/DynamicRangeProcessorHistoryPanel.cpp:42]: Agreed. Will solve when rebasing.

**Pull request merged**